### PR TITLE
Deprecate infrastructure-iot package

### PR DESCRIPTION
**Pull request by adamjhickey**
moving it to archive directory. Clean up references in various files.

### CODE REVIEW

File: common/config/rush/pnpm-lock.yaml [hunk 1]
```diff
@@ -58,7 +58,7 @@ importers:
       '@itwin/appui-abstract': 4.0.0_@itwin+core-bentley@4.0.0
       '@itwin/appui-layout-react': 4.1.0_1c726c265f9cde6743ef19e93af5aeaa
       '@itwin/appui-react': 4.1.0_c4fc7958281cb1014a865ba80498a059
-      '@itwin/breakdown-trees-react': file:../../packages/itwin/breakdown-trees_29ec31b409c64649c3a02bf8d7d1122e
+      '@itwin/breakdown-trees-react': file:../../packages/itwin/breakdown-trees_2a5c7b3246bd2f72b770f8e93a49e389
```
Comment (aruniverse) [inline common/config/rush/pnpm-lock.yaml:61]: im surprised how little this file changed. can you run `rush update` again and commit the modified lockfile if any?

Comment (adamjhickey) [reply common/config/rush/pnpm-lock.yaml:61]: I did a couple times to be sure. I can do it again.

Comment (adamjhickey) [reply common/config/rush/pnpm-lock.yaml:61]: No additional changes to the lock file after `rush update`.

**Comment by adamjhickey** [comment]
Just to confirm, the failing tests appear to match those failing in other PRs. I don't suspect any of my changes as the cause.

**Comment by aruniverse** [comment]
@adamjhickey the pipelines are failing due to a bad version of node.
Can you change the pipelines in `common/config/azure-pipelines` from `16.x` to `16.20.0` ?
@DanRod1999 can help as needed

**Review by AntonAI — APPROVED** [review]

File: common/config/azure-pipelines/ci.yaml [hunk 2]
```diff
@@ -21,14 +21,14 @@ jobs:
     condition: succeeded()
     strategy:
       matrix:
-        "Windows - Node 16":
-          NodeVersion: 16.x
+        "Windows - Node 16.20.0":
+          NodeVersion: 16.20.x
```
Comment (aruniverse) [inline common/config/azure-pipelines/ci.yaml:25]: youre going to need to pin it down to `16.20.0`; `16.20.1` is bad

Comment (adamjhickey) [reply common/config/azure-pipelines/ci.yaml:25]: Right. That was my first try but it didn't seem to like it... tacking another .x on the end

From failed pipeline:
"Starting: Use Node 16.20.0.x"

Comment (aruniverse) [reply common/config/azure-pipelines/ci.yaml:25]: need to update 
image

Comment (adamjhickey) [reply common/config/azure-pipelines/ci.yaml]: Thanks for the tip. Looking much better now.

**Pull request merged**